vyxal/transpile.py

```diff
@@ -0,0 +1,293 @@
+            )
+            + indent_str(
+                f"else: stack = pop(parameters, {struct.branches[0]}[0]"
+                + ", ctx)",
+                indent + 1,
+            )
+            + indent_str("ctx.context_values.append(stack[::])", indent + 1)
+            + indent_str("ctx.input_level += 1", indent + 1)
+            + indent_str(
+                "ctx.input_values[ctx.input_level] = [stack[::], 0]",
+                indent + 1,
+            )
+            + transpile_ast(struct.branches[1], indent + 1)
+            + indent_str("ret = pop(stack, ctx=ctx)", indent + 1)
+            + indent_str("ctx.context_values.pop()", indent + 1)
+            + indent_str("ctx.input_level -= 1", indent + 1)
+            + indent_str("return ret", indent + 1)
+            + indent_str(f"stack.append(_lambda_{signature})", indent)
+        )
+
+    if isinstance(struct, structure.FunctionReference):
+        return indent_str(f"stack.append(FN_{struct.branches[0]})", indent)
+
+    if isinstance(struct, structure.ListLiteral):
+        # We have to manually build this because we don't know how
+        # many list items there will be.
+
+        temp = indent_str("temporary_list = []", indent)
+        for x in struct.items:
+            temp += (
+                indent_str("def list_item(s, ctx):", indent)
+                + indent_str("stack = s[::]", indent + 1)
+                + transpile_ast(x, indent + 1)
+                + indent_str("return pop(stack, ctx=ctx)", indent + 1)
+                + indent_str(
+                    "temporary_list.append(list_item(stack, ctx)", indent
+                )
+            )
+
+        temp += indent_str("stack.append(temp_list[::]", indent)
+        return temp
+
+    if isinstance(struct, structure.MonadicModifier):
+        element_A = transpile_ast([lambda_wrap([struct.function_A])], indent)
+        return (
+            element_A + "\n" + elements.modifiers.get(struct.modifier, "pass")
+        )
+
+    if isinstance(struct, structure.DyadicModifier):
+        element_A = transpile_ast([lambda_wrap([struct.function_A])], indent)
+        element_B = transpile_ast([lambda_wrap([struct.function_B])], indent)
+        return (
+            element_A
+            + "\n"
+            + element_B
+            + "\n"
+            + indent_str(
+                elements.modifiers.get(struct.modifier, "pass"), indent
+            )
+        )
+    if isinstance(struct, structure.TriadicModifier):
+        element_A = transpile_ast([lambda_wrap([struct.function_A])], indent)
+        element_B = transpile_ast([lambda_wrap([struct.function_B])], indent)
+        element_C = transpile_ast([lambda_wrap([struct.function_C])], indent)
+        return (
+            element_A
+            + "\n"
+            + element_B
+            + "\n"
+            + element_C
+            + "\n"
+            + elements.modifiers.get(struct.modifier, "pass")
+        )
+
+    raise ValueError(struct)
```

Comment:
Is this error never supposed to be reached? If it is unreachable, use `assert False`; otherwise, add some more detail to the error message?